core/src/validator.rs

```diff
@@ -24,234 +17,217 @@ pub enum Error {
     #[error("WASM error: {0}")]
     Wasm(#[from] wasm::Error),
     /// Validator denied the operation.
-    #[error("Validator `{validator_id}` denied the operation `{operation}`: `{reason}`")]
+    #[error("Validator denied the operation `{operation}`: `{reason}`")]
     ValidatorDeny {
-        /// Validator ID.
-        validator_id: <Validator as Identifiable>::Id,
         /// Denial reason.
-        reason: DenialReason,
+        reason: data_model_validator::DenialReason,
         /// Denied operation.
-        operation: NeedsPermissionBox,
+        operation: data_model_validator::NeedsPermissionBox,
     },
 }
 
-/// Result type for [`Chain`] operations.
+/// Result type for [`Validator`] operations.
 pub type Result<T, E = Error> = core::result::Result<T, E>;
 
-/// Chain of *runtime* permission validators. Used to validate operations that require permissions.
+/// Validator that checks if user has permission to perform some operation.
 ///
-/// Works similarly to the
-/// [`Chain of responsibility`](https://en.wikipedia.org/wiki/Chain-of-responsibility_pattern).
-/// The validation of an operation is forwarded to all
-/// validators in the chain which have the required type.
-/// The validation stops at the first
-/// [`Deny`](iroha_data_model::permission::validator::Verdict::Deny) verdict.
+/// Can be upgraded with [`Upgrade`](iroha_data_model::isi::Upgrade) instruction.
 #[derive(Clone)]
```

Comment:
```suggestion
#[derive(DebugCustom, Clone)]
#[fmt = "Validator {...}")]
```

Reply:
I'm not forcing you to use `DebugCustom` but it seems to be a thing in our codebase